### PR TITLE
Bls12381

### PR DESCRIPTION
**Pull request by matteosz**
This PR introduces a different implementation of bls12381 (kilic) to kyber, which is faster than the actual circl implementation

### CODE REVIEW

**Comment by matteosz** [comment]
> I know this is a work in progress, but in general adding (a lot) more documentation would be helpful, as right now the code seems mostly undocumented.
> 
> It would also seem that `go fmt` wasn't run on all files. You might want to configure your editor to do this automatically.

Yep, I'll add some more doc. This was put on hold as I'm waiting the drand fork to be merged before working and merging this PR, to avoid further conflicts

File: pairing/bls12381/kilic/g1.go
Comment (AnomalRoil) [inline pairing/bls12381/kilic/g1.go:9]: Given the fact that https://github.com/kilic/bls12-381 is now archived, I'm starting to have second thoughts about including an adapter for it in Kyber.
WDYT @pierluca @ineiti ?

File: pairing/bls12381/circl/suite_test.go
Comment (AnomalRoil) [inline pairing/bls12381/circl/suite_test.go:19]: actually if this code is already in kyber/utils/test, couldn't we re-use the functions from there??

File: pairing/bn256/suite_test.go
Comment (AnomalRoil) [inline pairing/bn256/suite_test.go:410]: Why deleting it?

File: sign/test/bls_test.go
Comment (AnomalRoil) [inline sign/test/bls_test.go:6]: why not test both?

File: pairing/bn256/suite_test.go
Comment (matteosz) [reply pairing/bn256/suite_test.go:410]: Somehow it got deleted in the merge, not sure why

File: pairing/bls12381/kilic/g1.go
Comment (ineiti) [reply pairing/bls12381/kilic/g1.go:9]: You could:
- fork it into dedis/bls12-381 to avoid it missing if the author removes it
- effectively drop support for it

I have to admit I'm not sure if somebody is using it. But if it's not too big a hassle, I think the goal of kyber should be to have as many algorithms as possible.

So my proposition is to do nothing here - or perhaps add a comment somewhere and writing what you decided and how.

File: pairing/bls12381/bls12381_test.go
Comment (AnomalRoil) [reply pairing/bls12381/bls12381_test.go]: I've done it in a commit below, and it seems these tests are never being run @matteosz

File: pairing/bls12381/kilic/suite.go
Comment (matteosz) [reply pairing/bls12381/kilic/suite.go:88]: Yes, I'll do another PR for that

File: pairing/bls12381/bls12381_test.go
Comment (matteosz) [reply pairing/bls12381/bls12381_test.go]: Solved

File: pairing/bls12381/kilic/group.go
Comment (AnomalRoil) [reply pairing/bls12381/kilic/group.go:20]: Was that 👍🏻  a "yes we need to export it" or a "I'll make it a non-exported or a test-only interface"? xD

**Review by AnomalRoil — APPROVED** [review]
I think you just forgot to move the `circl/suite_test.go` file to its new folder, no?

File: pairing/bls12381/kilic/group.go
Comment (matteosz) [reply pairing/bls12381/kilic/group.go:20]: Yeah sorry forgot to remove it..it was used because kilic wasn't using the SubGroupElement used in kyber, so I just used that instead, now the GroupChecker can simply be removed instead of unexporting it :)

File: pairing/circl_bls12381/suite_test.go
Comment (matteosz) [reply pairing/circl_bls12381/suite_test.go:1]: Oh no, all the methods contained in that file were transferred to general `bls12381_test.go`, so there was no reason to keep the duplicated tests.

**Pull request merged**
